azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -433,6 +434,19 @@ public void updateProjectPermission(final Project project, final String name,
       eventData.put("updatedGroup", "null");
     }
 
+    //Getting updated user and Group permissions as String
+    Map<String, String> updatedUserPermissionMap = new HashMap<>();
```

Comment:
Use final keyword wherever applicable.
Initialize using the size of respective structures.

Reply:
Ack.
Made the change, kindly have a look